lib/TWCManager/TWCMaster.py

```diff
@@ -70,11 +74,21 @@ class TWCMaster:
 
     def __init__(self, TWCID, config):
         self.config = config
-        self.debugLevel = config["config"]["debugLevel"]
+        self.debugLevel = config["config"].get("debugLevel", 1)
+        self.debugOutputToFile = config["config"].get("debugOutputToFile", False)
         self.TWCID = TWCID
         self.subtractChargerLoad = config["config"]["subtractChargerLoad"]
         self.advanceHistorySnap()
 
+        if self.debugOutputToFile == True:
+            self.logger = logging.getLogger("TWCManager")
+            self.logger.setLevel(logging.INFO)
+            handler = TimedRotatingFileHandler(
+                "/etc/twcmanager/twcmanager.log", when="H", interval=1, backupCount=24
+            )
+            handler.setFormatter(logging.Formatter("%(asctime)s - %(message)s"))
+            self.logger.addHandler(handler)
+
```

Comment:
We've already introduced the idea of Logging modules, which can log messages in various formats such as CSV, text or to a database. The intention has always been to move to these modules - my reservations about this are on a number of levels - I've always avoided unnecessary writes to files because people are using this on RPIs with SD cards, hence why we try to be judicious - I understand that it will clean up after itself but it will also write constantly to the SD card. Options such as MySQL or Console logging avoid this, and I was planning on adding Syslog too - ways that people can offload the logging from the RPI device to avoid killing their SD card. Perhaps this could just be a new logging module - we don't have a text output logging module yet. Note that the logging module doesn't yet log all messages but there's no reason that can't be introduced.

Reply:
I saw that, looked at it - and was convinced that you don't want to log this console output debug thing. ;-) I also agree with your intervention to not kill the SDCard - that's why I added an additional parameter - so it's only for debugging. I will look into the logging modules to find another way there...

Reply:
Thanks, the new module you've put together looks good! It is also entirely optional and fits in with the architecture, so it's a win win in my view.